### DISCUSSION TITLE
Can't Login After Update AtroPIM 1.0 to 1.2.5

### DISCUSSION THREAD

**Discussion by marcus-at-localhost** [atrocore/atrocore#182] [Q&A] [answered]
Hey, 
I installed AtroPIM in January, I believe it was version 1.0 and now I'm trying to get to the latest version 1.2.5 (atrocore) to run.

What I did was overwriting the composer.json and composer.phar and ran `php composer.phar update`

Installation looked ok, but I can't login anymore (wrong username/password)
I tried to compare the config file and I believe there were some changes in the `authenticationMethod` for example, but it's just too much keys that aren't self-explainory. I fear I bricked my database and all the custom relations and stuff.

In case I have to do a clean install, is there a way to take over that database? Because reinstalling seems to clear all tables.

**Comment by rratsun** [comment]
Hello,
Can you give us an update log?

**Reply by marcus-at-localhost** [reply]
for some reason, data/logs is empty. or where do I find the log? 
I can tell, that login attempts ends in a 401

**Comment by atrolex** [comment]
The right way to update the system is to use the Module Manager. For now, it is hard to tell you exactly where the problem is. It can take us a lot of time, so we can offer you paid service to solve your problem. Otherwise, you should restore the system from backup and do update via Module Manager. It is important that all database migrations jobs are done.

**Comment by marcus-at-localhost** [comment, marked as answer]
I think I boiled it down to this change: https://github.com/atrocore/atrocore/commit/d0ad0734cb6987d73c6c844d80a4db7f3c53cbf0#diff-394754d5e2a5126d9b1d631a10e5bba6bfedbdb548d4cf5586175dcaaa83113d

I have no `PHP_AUTH_USER` in my headers (I do have `$_SERVER["REDIRECT_HTTP_AUTHORIZATION"]=> "Basic YWRtaXXXXXWRtaW4="` when I use `SetEnvIf Authorization .+ HTTP_AUTHORIZATION=$0` in my htaccess (Server runs PHP 7.3 CGI/FastCGI)

I went through several topics like this: https://stackoverflow.com/questions/11990388/request-headers-bag-is-missing-authorization-header-in-symfony-2 but I can't make `PHP_AUTH_USER` show up.

Is that something you could consider to work around? Is there something else I should try?

**Reply by marcus-at-localhost** [reply]
I got in with adding this:

```php
    public function call()
    {
        $req = $this->app->request();

        $uri = $req->getResourceUri();
        $httpMethod = $req->getMethod();

        if (isset($_SERVER['REDIRECT_HTTP_AUTHORIZATION'])) {
           list($authUsername, $authPassword) = explode(':' , base64_decode(substr($_SERVER['REDIRECT_HTTP_AUTHORIZATION'], 6)));
        }else{
            $authUsername = $req->headers('PHP_AUTH_USER');
            $authPassword = $req->headers('PHP_AUTH_PW');
        }
     }
```

It's not working 100% though.

**Reply by marcus-at-localhost** [reply]
Ok I think I solved it. I ran the composer update again and now all menus and items show up again.

Do you want me to create an issue in the repo to work around missing `PHP_AUTH_USER` headers or I can create a pull request, but you might know better further implications of changing that.
